wasm/src/graph_traversal.rs

```diff
@@ -15,24 +15,132 @@ use crate::{
     },
 };
 
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct TraversalOptions {
+    entry_nodes: Vec<String>,
+    // if this is None, all edge types will be traversed
+    edge_types: Option<Vec<String>>,
+    max_depth: u32,
+    /// if true, multiple traversals - one for each edge type - will be performed and the results will be combined
+    /// if false, one traversal over all edge types will be performed
+    separate_edges: bool,
+}
+
+#[wasm_bindgen]
+impl TraversalOptions {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        entry_nodes: Vec<String>,
+        edge_types: Option<Vec<String>>,
+        max_depth: u32,
+        separate_edges: bool,
+    ) -> TraversalOptions {
+        TraversalOptions {
+            entry_nodes,
+            edge_types,
+            max_depth,
+            separate_edges,
+        }
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn entry_nodes(&self) -> Vec<String> {
+        self.entry_nodes.clone()
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn edge_types(&self) -> Option<Vec<String>> {
+        self.edge_types.clone()
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn max_depth(&self) -> u32 {
+        self.max_depth
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn separate_edges(&self) -> bool {
+        self.separate_edges
+    }
+
+    #[wasm_bindgen(js_name = toString)]
+    pub fn to_string(&self) -> String {
+        format!("{:?}", self)
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct Path {
+    edges: Vec<EdgeStruct>,
+}
+
+#[wasm_bindgen]
+impl Path {
+    pub fn length(&self) -> usize {
+        self.edges.len()
+    }
+
+    pub fn truncate(&mut self, limit: usize) {
+        self.edges.truncate(limit);
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn edges(&self) -> Vec<EdgeStruct> {
+        self.edges.clone()
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn reverse_edges(&self) -> Vec<EdgeStruct> {
+        self.edges.iter().rev().cloned().collect()
+    }
+
+    pub fn equals(&self, other: &Path) -> bool {
+        self.edges == other.edges
```

Comment:
Haha I'm so used to JS weirdness.. does this do a full deep equality check?

Reply:
yes, `==` is based on the `PartialEq` trait, which checks for deep equality (at here, technically the trait could be implemented in an arbitrary way)